server/commands/check.go

```diff
@@ -415,36 +419,65 @@ func (query *CheckQuery) resolveTupleToUserset(ctx context.Context, rc *resoluti
 			}
 			return serverErrors.HandleError("", err)
 		}
+
 		if rc.shouldShortCircuit() {
 			break // the user was resolved already, avoid launching extra lookups
 		}
 
-		uObject, uRelation := tupleUtils.SplitObjectRelation(tuple.GetUser())
-		// We only proceed in the case that userRelation == userset.GetComputedUserset().GetRelation().
-		// uRelation may be empty, and in this case we set it to userset.GetComputedUserset().GetRelation().
-		if uRelation == "" {
-			uRelation = node.TupleToUserset.GetComputedUserset().GetRelation()
+		userObj, userRel := tupleUtils.SplitObjectRelation(tuple.GetUser())
+		usersetRel := node.TupleToUserset.GetComputedUserset().GetRelation()
+
+		// userRel may be empty, and in this case we set it to usersetRel.
+		if userRel == "" {
+			userRel = usersetRel
 		}
-		if uRelation != node.TupleToUserset.GetComputedUserset().GetRelation() {
+		// We only proceed in the case that userRel == usersetRel (=node.TupleToUserset.GetComputedUserset().GetRelation()).
```

Comment:
A couple of renames and move the comment to the line it is commenting on to add in readability.